test/hotspot/jtreg/compiler/macronodes/TestTopInMacroElimination.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Regression test for an assert triggered during allocation elimination because top is found during
+ * constructing new phis.
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib /
+ * @run main/othervm -XX:-ProfileExceptionHandlers compiler.macronodes.TestTopInMacroElimination
+ */
+
+package compiler.macronodes;
+
+public class TestTopInMacroElimination {
+    public static class A {
+        int a;
+    }
+
+    public static void main(String[] strArr) {
+        int i2 = 0;
+        for (int i = 0; i < 50; ++i)
+            try {
+                synchronized (new A()) {
+                    synchronized (TestTopInMacroElimination.class) {
+                        for (int var19 = 0; var19 < Integer.valueOf(i2);) {
+                            Integer.valueOf(var19);
+                        }
+                    }
+                }
+                for (int var8 = 0; var8 < 10000; ++var8) ;
+            } catch (ArithmeticException a_e) {
+            }
```

Comment:
I suggest to add braces for the `for` loop:
```suggestion
        for (int i = 0; i < 50; ++i) {
            try {
                synchronized (new A()) {
                    synchronized (TestTopInMacroElimination.class) {
                        for (int var19 = 0; var19 < Integer.valueOf(i2);) {
                            Integer.valueOf(var19);
                        }
                    }
                }
                for (int var8 = 0; var8 < 10000; ++var8) ;
            } catch (ArithmeticException a_e) {
            }
        }
```